File: app/src/main/java/com/mnowak/cirriculumvitae/feature/moreInfo/MoreInfoPagerAdapter.java
package com.mnowak.cirriculumvitae.feature.moreInfo;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;

import java.util.List;

public class MoreInfoPagerAdapter extends FragmentStatePagerAdapter {

    private final List<Fragment> pages;

    public MoreInfoPagerAdapter(FragmentManager fm, List<Fragment> pages) {
        super(fm);
        this.pages = pages;
    }

    @Override
    public int getCount() {
        return pages.size();
    }

    @Override
    public Fragment getItem(int position) {
        return pages.get(position);
    }

}

File: app/src/main/java/com/mnowak/curriculumvitae/feature/moreInfo/MoreInfoPagerAdapter.kt
package com.mnowak.curriculumvitae.feature.moreInfo

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter

class MoreInfoPagerAdapter(fm: FragmentManager, private val pages: List<Fragment>) : FragmentStatePagerAdapter(fm) {

    override fun getCount(): Int = pages.size

    override fun getItem(position: Int): Fragment = pages[position]
}
